lib/aws/rails/sesv2_mailer.rb

```diff
@@ -0,0 +1,64 @@
+# frozen_string_literal: true
+
+require 'aws-sdk-sesv2'
+
+module Aws
+  module Rails
+    # Provides a delivery method for ActionMailer that uses Amazon Simple Email
+    # Service V2.
+    #
+    # Once you have an SESv2 delivery method you can configure Rails to
+    # use this for ActionMailer in your environment configuration
+    # (e.g. RAILS_ROOT/config/environments/production.rb)
+    #
+    #     config.action_mailer.delivery_method = :sesv2
+    #
+    # Uses the AWS SDK for Ruby's credential provider chain when creating an SESV2
+    # client instance.
+    class Sesv2Mailer
+      # @param [Hash] options Passes along initialization options to
+      #   [Aws::SESV2::Client.new](https://docs.aws.amazon.com/sdk-for-ruby/v3/api/Aws/SESV2/Client.html#initialize-instance_method).
+      def initialize(options = {})
+        @client = SESV2::Client.new(options)
+      end
+
+      # Rails expects this method to exist, and to handle a Mail::Message object
+      # correctly. Called during mail delivery.
+      def deliver!(message)
+        send_opts = {}
+        send_opts[:content] = {}
+        send_opts[:content][:raw] = { data: message.to_s }
+
+        # Check for Rails >=6 compatibility
+        send_opts[:from_email_address] = if message.respond_to?(:from_address)
+                                           message.from_address&.to_s
+                                         else
+                                           Array.wrap(message.from).first&.to_s
+                                         end
```

Comment:
Do you have any documentation or information on this that I can look into? I am assuming this is Mail::Message

Reply:
Hi, yes they added some extensions for Mail::Message in rails, you can see them here:
https://github.com/rails/rails/blob/ade27e5028ccbaaa5db8d966500844017925a751/actionmailbox/lib/action_mailbox/mail_ext/addresses.rb

I can only see those extensions from rails v6 onwards, and I noticed it just because the rails app I'm maintaining is still on v5.2.7.1, and threw an error when trying to read "message.from_address". I could add the extensions just to our app, but I thought that simple check might make this gem backwards compatible with Rails 5.2.

Reply:
Thanks. What about `message.from` and why doesn't this work generically for any Rails version, even newer ones?

Reply:
I think that `message.from` should be ok to use from looking at the comments on the code for [mail/message](https://github.com/mikel/mail/blob/bf02b6772e4df20e165c008f595f4eadb8c595a7/lib/mail/message.rb#L658).
But I am not very familiar with the `mail` gem, so please take a look and let me know any thoughts. 
I'm happy to implement the changes an update this PR. My suggestion is to go with `message.from`. I just put that check as the other PR #74 used `message.from_address`, but actually I think we can replace that with `message.from`.

Reply:
Yeah, I think message.from might work generically and we won't need to branch here. That makes sense that you carried it over originally. Can you confirm with your testing (assuming you are doing this locally)?